packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.native.test.tsx

```diff
@@ -24,4 +25,51 @@ describe('<BaseButton />', () => {
       }
     }
   });
+  it('should render large size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="large">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render medium size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="medium">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render small size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="small">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render xsmall size button', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton size="xsmall">{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon without text', () => {
+    const { toJSON } = renderWithTheme(<BaseButton icon={CreditCardIcon} />);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon with default iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(<BaseButton icon={CreditCardIcon}>{buttonText}</BaseButton>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon with left iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { toJSON } = renderWithTheme(
+      <BaseButton iconPosition="left" icon={CreditCardIcon}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render button with icon with right iconPosition', () => {
```

Comment:
do we want to write tests for all permutation combinations? I wonder that's just too much to bloat up things. I think we discussed something similar last time. don't remember the conclusion 🤔

Reply:
Don't remember either 😅 But if we don't ensure tests for at least the commonly used combinations, it would be open to breakage

Reply:
That's too many snapshots. :( 

We can reduce some of them. 

1. For icons we should instead do testing by checking if icon exists in the document. 

Eg: 

```diff
+    expect(getByTestId('icon-left')).toBeInTheDocument()
+    expect(getByTestId('icon-right')).not.toBeInTheDocument()
-    expect(container).toMatchSnapshot();

+    expect(getByTestId('icon-right')).toBeInTheDocument()
+    expect(getByTestId('icon-left')).not.toBeInTheDocument()
-    expect(container).toMatchSnapshot();
```

2. for button type prop, just assert the type prop, don't need snapshot

```ts
    expect(getByRole('button')).toHaveAttribute('type', 'reset');
```

Reply:
> Would want to have snapshot test for icon because padding of text changes based on icon presence and position.

I feel like thats a bit overkill just for testing padding. Rather we can test individually for the padding by accessing the computed styles.

Reply:
What's the harm doing a snapshot test? Lets relook at this if our tests feel bloated? Right now it seems fine